### PR TITLE
Takashi

### PR DESCRIPTION
**Pull request by takashim0101**

### CODE REVIEW

**Comment by takashim0101** [comment]
## Feature: Implement API 1 (Car Valuation) with Test-Driven Development (TDD)

This pull request introduces the complete implementation of API 1, responsible for calculating car values based on the "Magic Calculator" business rules, following a Test-Driven Development (TDD) approach.

### Key Changes:

- **API 1 Logic (`api1_car_value/carValuation.js` - *Note: Currently consolidated in `server.js` for this PR's context*):**
    - Implemented the `calculateCarValue` function, which computes car value by summing alphabetical positions of the model name (A=1, Z=26), multiplying by 100, and adding the year.
    - Includes robust input validation for `model` (non-empty string) and `year` (positive integer).
    - Utilizes a custom `APIError` class for standardized error handling.
    - Helper functions (`calculateAlphabetSum`, `validateCarValuationInput`) were created for improved modularity and readability.

- **API Endpoint Integration (`server.js`):**
    - Configured the `/api/car-value` POST endpoint to utilize the `calculateCarValue` logic.
    - Implemented `try-catch` blocks for comprehensive error handling, returning appropriate HTTP status codes (400 for invalid input, 500 for unexpected errors).

- **Comprehensive Test Suite (`tests/api-1.test.js`):**
    - Developed a detailed set of unit tests for `calculateCarValue` and integration tests for the `/api/car-value` endpoint.
    - Tests cover normal cases, various edge cases (e.g., single-letter models, mixed casing, special characters), and all specified error conditions, ensuring high test coverage.

### Why these changes are important:

- **Fulfills Mission Requirements:** Directly addresses Task 5 (Create Test Cases), Task 6 (Create Unit Tests), and Task 7 (Create API) for API 1.
- **TDD Adherence:** The implementation strictly followed the TDD cycle, with tests written before the corresponding code, ensuring functionality and robustness.
- **Robustness:** The API now handles valid inputs correctly and provides clear error messages for invalid or missing data.
- **Readiness for Refactoring:** While currently consolidated in `server.js` for simplicity in this PR, the internal structure of `calculateCarValue` is designed for easy extraction into separate modules (`carValuation.js`, `carValueRouter.js`) in a future refactoring phase (Task 8).

### How to Test:

1.  Ensure all dependencies are installed (`npm install`).
2.  Start the server (`npm start`).
3.  Run the API 1 specific tests: `npm run test:api1`
4.  Run all tests: `npm test`

Looking forward to your feedback and review!

**Comment by takashim0101** [comment]
**feat: Implement and test API 1 for car value calculation**

This commit completes the implementation of API 1 (`/api/car-value`).

**Key changes include:**
- The endpoint now supports both single car object input and an array of car objects.
- Each car object in an array is processed, returning either its calculated carValue or an error.
- Comprehensive integration tests have been added to `tests/api-1.test.js` to cover both single and array input scenarios, including positive, edge, and negative cases.

_This allows for flexible car value calculation for individual vehicles or batches._

File: backend-m2/server.js [hunk 1]
```diff
@@ -1,20 +1,199 @@
...
+ * Non-letters are ignored, like magic!
+ * @param {string} modelName - The car model name (like "Civic").
+ * @returns {number} The total "alphabet power" number.
+ */
+function calculateAlphabetSum(modelName) {
+    let sum = 0; // We start counting from zero.
+    const baseCharCodeA = 'A'.charCodeAt(0); // This is the secret number for the letter 'A'.
+
+    // We look at each letter in the car's name.
+    for (let i = 0; i < modelName.length; i++) {
```
Comment (Realist2022) [inline backend-m2/server.js:43]: You could shorten your code like this if you wanted..? :
    for (let i = 0; i < modelName.length; i++) {
        const char = modelName[i].toUpperCase();
        if (char >= 'A' && char <= 'Z') {
            sum += (char.charCodeAt(0) - baseCharCodeA + 1);
